examples/daal/cpp/source/k_nearest_neighbors/kdtree_knn_dense_search_batch.cpp

```diff
@@ -0,0 +1,198 @@
+      kdtree_knn_classification::computeIndicesOfNeighbors;
+
+  /* Train the KD-tree based kNN model */
+  algorithm.compute();
+  /* Retrieve the results of the training algorithm  */
+  trainingResult = algorithm.getResult();
+}
+
+void testModel(
+    kdtree_knn_classification::training::ResultPtr &trainingResult,
+    kdtree_knn_classification::prediction::ResultPtr &predictionResult) {
+  /* Initialize FileDataSource<CSVFeatureManager> to retrieve the test data from
+   * a .csv file */
+  FileDataSource<CSVFeatureManager> testDataSource(
+      testDatasetFileName, DataSource::notAllocateNumericTable,
+      DataSource::doDictionaryFromContext);
+
+  /* Create Numeric Tables for testing data and labels */
+  NumericTablePtr testData =
+      HomogenNumericTable<>::create(nFeatures, 0, NumericTable::doNotAllocate);
+  NumericTablePtr testGroundTruth =
+      HomogenNumericTable<>::create(1, 0, NumericTable::doNotAllocate);
+  NumericTablePtr mergedData(new MergedNumericTable(testData, testGroundTruth));
+
+  /* Retrieve the data from input file */
+  testDataSource.loadDataBlock(mergedData.get());
+
+  /* Create algorithm objects for KD-tree based kNN prediction with the default method
+   */
+  kdtree_knn_classification::prediction::Batch<> algorithm;
+
+  /* Pass the testing data set and trained model to the algorithm */
+  algorithm.input.set(classifier::prediction::data, testData);
+  algorithm.input.set(classifier::prediction::model,
+                      trainingResult->get(classifier::training::model));
+  algorithm.parameter.k = kNeighbors;
+  algorithm.parameter.nClasses = nClasses;
+  algorithm.parameter.resultsToCompute =
+      kdtree_knn_classification::computeDistances |
+      kdtree_knn_classification::computeIndicesOfNeighbors;
+
+  /* Compute prediction results */
+  algorithm.compute();
+
+  /* Retrieve algorithm results */
+  predictionResult = algorithm.getResult();
+}
+
+void readGroundTruth(NumericTablePtr &groundTruthIndices,
+                     NumericTablePtr &groundTruthDistances) {
+  /* Initialize FileDataSource<CSVFeatureManager> to retrieve the test data from
+   * a .csv file */
+  FileDataSource<CSVFeatureManager> indicesDataSource(
+      groundTruthIndicesFileName, DataSource::notAllocateNumericTable,
+      DataSource::doDictionaryFromContext);
+  FileDataSource<CSVFeatureManager> distancesDataSource(
+      groundTruthDistancesFileName, DataSource::notAllocateNumericTable,
+      DataSource::doDictionaryFromContext);
+
+  /* Create Numeric Tables for testing data and labels */
+  groundTruthIndices =
+      HomogenNumericTable<>::create(kNeighbors, 0, NumericTable::doNotAllocate);
+  groundTruthDistances =
+      HomogenNumericTable<>::create(kNeighbors, 0, NumericTable::doNotAllocate);
+
+  /* Retrieve the data from input file */
+  indicesDataSource.loadDataBlock(groundTruthIndices.get());
+  distancesDataSource.loadDataBlock(groundTruthDistances.get());
+}
```

Comment:
Why separate function is required?

Reply:
Probably to differentiate data types. In SYCL example the same approach is used:
`printNumericTable<int>` in `printIndicesResults` and `printNumericTable<>` in `printDistancesResults`. It is the way to work around weird output like `0.000` instead of just `0` in output of indices.